pyro/distributions/torch_distribution.py

```diff
@@ -82,22 +111,14 @@ def expand(self, batch_shape):
         :return: An expanded version of this distribution.
         :rtype: :class:`ReshapedDistribution`
         """
-        batch_shape = list(batch_shape)
-        if len(batch_shape) < len(self.batch_shape):
-            raise ValueError("Expected len(batch_shape) >= len(self.batch_shape), "
-                             "actual {} vs {}".format(len(batch_shape), len(self.batch_shape)))
-        # check sizes of existing dims
-        for dim in range(-1, -1 - len(self.batch_shape), -1):
-            if batch_shape[dim] == -1:
-                batch_shape[dim] = self.batch_shape[dim]
-            elif batch_shape[dim] != self.batch_shape[dim]:
-                if self.batch_shape[dim] != 1:
-                    raise ValueError("Cannot broadcast dim {} of size {} to size {}".format(
-                        dim, self.batch_shape[dim], batch_shape[dim]))
-                else:
-                    raise NotImplementedError("https://github.com/uber/pyro/issues/1119")
-        sample_shape = batch_shape[:len(batch_shape) - len(self.batch_shape)]
-        return self.expand_by(sample_shape)
+        try:
+            sample_shape = self._sample_shape(batch_shape)
+            return self.expand_by(sample_shape)
+        except ShapeMismatchError:
+            raise NotImplementedError("`TorchDistributionMixin.expand()` cannot expand " +
```

Comment:
nit: no need for plus signs here, it's cheaper to concatenate the strings in parsing than in execution.